code/modules/bitrunning/objects/gimmick_disks/dungeon_disk.dm

```diff
@@ -0,0 +1,222 @@
+		/datum/reagent/sodium,
+		/datum/reagent/stable_plasma,
+		/datum/reagent/consumable/sugar,
+		/datum/reagent/sulfur,
+		/datum/reagent/toxin/acid,
+		/datum/reagent/water,
+		/datum/reagent/fuel,
+		/datum/reagent/toxin/leadacetate, // Funny chems
+		/datum/reagent/consumable/liquidgibs,
+		/datum/reagent/consumable/nutriment,
+		/datum/reagent/consumable/coffee,
+	)
+
+/obj/item/reagent_containers/cup/bottle/alchemist_basic/Initialize(mapload)
+	list_reagents = list(pick(possible_reagents) = 50)
+	. = ..()
+
+/obj/item/reagent_containers/cup/bottle/alchemist_random
+	name = "skull-labeled bottle"
+	desc = "A small bottle. You don't remember what you put in it."
+
+	/// List of random names this may have
+	var/static/list/possible_names = list(
+		"unlabeled bottle",
+		"skull-labeled bottle", // Labels
+		"heart-labeled bottle",
+		"explosion-labeled bottle",
+		"fish-labeled bottle",
+		"smiley-labeled bottle",
+		"frown-labeled bottle",
+		"interrobang-labeled bottle",
+		"plus-labeled bottle",
+		"d20-labeled bottle",
+		"unreadably-labeled bottle",
+		"black-labeled bottle",
+		"empty-labeled bottle",
+		"age-yellowed bottle", // Qualities
+		"blood-tinged bottle",
+		"ash-marred bottle",
+		"claw-scratched bottle",
+		"marker-marked bottle",
+		"cracked bottle",
+		"ominous bottle", // Abstract
+	)
+
+/obj/item/reagent_containers/cup/bottle/alchemist_random/Initialize(mapload)
+	list_reagents = list(get_random_reagent_id() = rand(20, 50))
+	. = ..()
+	name = pick(possible_names)
+
+/obj/item/storage/box/alchemist_basic_chems
+	name = "box of alchemical bases"
+	desc = "Contains a set of basic reagents, for all your potion-making needs! If only you labeled them."
+	illustration = "beaker"
+
+/obj/item/storage/box/alchemist_basic_chems/PopulateContents()
+	for(var/i in 1 to 7)
+		if(prob(1))
+			new /obj/item/reagent_containers/cup/glass/coffee(src)
+			continue
+		new /obj/item/reagent_containers/cup/bottle/alchemist_basic(src)
+
+/obj/item/storage/box/alchemist_random_chems
+	name = "box of potions"
+	desc = "An especially fancy box to keep your finished potions safe."
+	icon_state = "syndiebox"
+	illustration = "beaker"
+
+/obj/item/storage/box/alchemist_random_chems/PopulateContents()
+	for(var/i in 1 to 7)
+		if(prob(1))
+			new /obj/item/reagent_containers/cup/glass/coffee(src)
+			continue
+		new /obj/item/reagent_containers/cup/bottle/alchemist_random(src)
+
+/obj/item/storage/box/alchemist_chemistry_kit
+	name = "box of alchemy tools"
+	desc = "Contains everything needed for the up and coming chemistry student to enact hazardous chemical mishaps in the comfort of their own home."
+
+/obj/item/storage/box/alchemist_chemistry_kit/PopulateContents()
+	new /obj/item/reagent_containers/cup/mortar(src)
+	new /obj/item/pestle(src)
+	new /obj/item/lighter/skull(src)
+	new /obj/item/ph_booklet(src)
+	new /obj/item/thermometer(src)
+	new /obj/item/storage/test_tube_rack/full(src)
+	new /obj/item/reagent_containers/cup/glass/coffee(src)
+
+
```

Comment:
I've given up keep your secrets

Reply:
Whenever I look at a large file with equal spacing between every part my eyes glaze over, so when I code stuff myself I use double spaces to section off distinct segments (i.e. different types of disks, or in big components definition/registry/init/destroy/such vs actual functionality)